### DISCUSSION TITLE
Amazon Q changes formatting of VS Code `settings.json`

### DISCUSSION THREAD

**Discussion by injust** [aws/codewhisperer-command-line-discussions#229] [Known bugs]
Every time Amazon Q upgrades, it changes my VS Code `settings.json` indentation from 4 spaces to 2, and it removes the trailing newline.

It used to actually change your settings (I think it was reverting it to an older version?), but it looks like that's fixed.

**Comment by grant0417** [comment]
This is for the Amazon Q for command line produce, see https://github.com/aws/aws-toolkit-vscode for support for the vscode extension.

**Reply by injust** [reply]
@grant0417 Not sure why this was closed -- I'm in the correct repo and have never installed https://github.com/aws/aws-toolkit-vscode.

Amazon Q for CLI is messing up my VS Code `settings.json`.

**Reply by grant0417** [reply]
Sorry, must have misunderstood what is happening here. Will try to investigate why this is happening.

**Reply by injust** [reply]
This should be easy to reproduce -- switch to 4 spaces indentation in VS Code's `settings.json`. Then uninstall the "Amazon Q for command line" VS Code extension, then kill and relaunch Amazon Q.
